src/extension.ts

```diff
@@ -153,7 +154,16 @@ async function activateServerForFolder(context: ExtensionContext, uri: Uri, fold
       return;
     }
   } catch (e) {
-    if (e instanceof Error) {
+    if (e instanceof MissingToolError) {
+      const link = e.installLink();
+      if (link) {
+        if (await window.showErrorMessage(e.message, `Install ${e.tool}`)) {
+          env.openExternal(link);
+        }
+      } else {
+        await window.showErrorMessage(e.message);
+      }
+    } else if (e instanceof Error) {
```

Comment:
Here at the top-level of extension init we catch all promise errors and also restore the `MissingToolError` with link popup.